galley/pkg/metadata/types.go

```diff
@@ -186,6 +186,9 @@ var (
 	// istio/networking/v1alpha3/sidecars metadata
 	IstioNetworkingV1alpha3Sidecars resource.Info
 
+	// istio/networking/v1alpha3/synthetic/serviceentries metadata
+	IstioNetworkingV1alpha3SyntheticServiceentries resource.Info
```

Comment:
(not critical) - I remember we decided to use annotations, wouldn't be cleaner to just put an annotation on the 
service entry ? The reason we introduced synthetic entries was that we didn't have (or considered) annotations.
Given that ingress and other configs may at some point be 'synthetic' - if it's not too late we might still change.

Reply:
We actually do both currently: the annotations exist on the ServiceEntry **and** we have synthetic ServiceEntries as a separate collection.  The collection provides system-wide feature separation, while the annotations are available to aid in quickly identifying what changed (i.e. service vs endpoints).

If we're saying that Galley integration in Pilot is generally an alpha feature, however, perhaps it's ok to mingle synthetic and non-synthetic into the same Galley collection (i.e. same snapshot). The client for all ServiceEntries in that case would have to use incremental MCP.

cc @ozevren @ayj

Reply:
If it's not too late - it may be worth choosing one and sticking with it. 

The 'synthetic' was based on the idea that any change in the 'synthetic' collections can be identified by the collection name without needing annotations or other processing. And my assumption was that we'll have a 'synthetic' type that will send only Endpoint changes, which was the problem we were trying to solve. 

But if annotations are introduced - we may not need the extra complexity, any entry ( like VirtualService derived from Ingress) can use an annotation and it's cleaner.

Reply:
Not really - we're saying that ServiceEntries for remote clusters will use MCP. Those are MCP clients that connect to the remote clusters - and only the remote clusters need to use incremental MCP.

The primary cluster will keep using standard MCP and not send synthetic ServiceEntry in 1.2 ( and until
we have sufficient testing and maturity ).